### ISSUE TITLE
Password should be required when changing email address

### ISSUE THREAD

**Issue by ScopeyNZ** [silverstripe/silverstripe-framework#8599]
## Affected Version

All

## Description

Currently a user must enter their existing password to change their password. You do not need to enter your password to change your email address. This is a bit of an issue because someone can now alter a password by changing the email address and using the "forgot password" functionality

## Steps to Reproduce

- Login to the CMS
- Change the email address
- Observe you don't have to enter your password to change your email address

## Notes

This is related to the idea of #8595 . GitHub has this as one of the actions that can only be done while you're in this heightened state of trust - and I think it makes good sense.

I've formatted this like a bug report but I'm happy to accept that it's an enhancement (and have left the triaging for someone else because I'm indifferent)

**Comment by robbieaverill** [comment]
I think it'd be an enhancement too. 

We should provide extension points so that something like an MFA module might be able to let you "press your security key" instead of entering your password too.

**Comment by elliot-sawyer** [comment]
@clarkepaul how would you expect the UX to work on this?

**Comment by maxime-rainville** [comment]
@clarkepaul On a related note, do we have a story somewhere for implementing a generic modal? It doesn't seem like we have consistent UX for displaying Modals. Hell, we're still using the browser's native `confirm()` method for deleting/archiving records.

**Comment by elliot-sawyer** [comment]
I've been meaning to ask this, I thought it was a new feature in 4.3 and couldn't find anything on how to trigger it!

**Comment by ScopeyNZ** [comment]
Considering there's a form field you could `FormBuilderModal`...

Otherwise I'd just use Reactstrap `Modal` and then ask @clarkepaul what we need to do to make it suitable.

**Comment by clarkepaul** [comment]
Yeah this should be the ReactStrap default modal, but we also use modals in the current CMS UI (insert media,  add to campaign etc.) We weren't sure about what flexibility we had with having the action in/not-in an action toolbar so that would need to be discussed when it comes to development.
